tests/integration/cmd_devfile_push_test.go

```diff
@@ -0,0 +1,156 @@
+package integration
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/openshift/odo/tests/helper"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("odo devfile push command tests", func() {
+	var project string
+	var context string
+	var currentWorkingDirectory string
+
+	var sourcePath = "/projects"
+
+	oc = helper.NewOcRunner("oc")
+
+	// This is run after every Spec (It)
+	var _ = BeforeEach(func() {
+		SetDefaultEventuallyTimeout(10 * time.Minute)
+		project = helper.CreateRandProject()
+		context = helper.CreateNewContext()
+		currentWorkingDirectory = helper.Getwd()
+		helper.Chdir(context)
+		os.Setenv("ODO_EXPERIMENTAL", "true")
```

Comment:
Do not use it like this because odo test spec (It) runs on two ginkgo test node in parallel, so it may cause race condition. To avoid such situation we have a env variable which can control preference path.

You can use 
```
os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
```
and in the spec i mean in the It block you can update the preference value. For example
```
helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
```

Reply:
I was wondering about the tests running in parallel and whether or not that would cause problems, good to know! I'll update my PR to include that.